supabase/functions/cron_email/index.ts

```diff
@@ -0,0 +1,154 @@
+import { serve } from 'https://deno.land/std@0.199.0/http/server.ts'
+import { getEnv, methodJson, sendRes } from '../_utils/utils.ts'
+import type { BaseHeaders } from '../_utils/types.ts'
+import { trackEvent } from '../_utils/plunk.ts'
+import { supabaseAdmin } from '../_utils/supabase.ts'
+
+// This is required propoably due to this https://github.com/supabase/postgrest-js/issues/408
+interface AppWithUser {
+  app_id: string
+  name: string | null
+  user_id: {
+    email: string
+  }
+}
+
+const thresholds = {
+  // Number of updates in plain number
+  updates: [
+    100,
+    1000,
+    10000,
+  ],
+  // Percentage in decimal form (0.9 ==== 90%)
+  failRate: [
+    0.80,
+    0.90,
+    0.95,
+  ],
+  // Number of app opens in plain number
+  appOpen: [
+    500,
+    1500,
+    5000,
+  ],
+}
+
+const funComparisons = {
+  updates: [
+    'a cupcake to every student in a small school!',
+    'a pizza to every resident of a small town!',
+    'a burger to everyone in a big city!',
+  ],
+  failRate: [
+    'Even cats don\'t land on their feet that often!',
+    'That\'s a success rate higher than the average pass rate of a tough university exam!',
+    'That\'s a success rate that even the best basketball players would envy!',
+  ],
+  appOpen: [
+    'Your app was opened more times than a popular local bakery\'s door!',
+    'Your app was more popular than the latest episode of a hit TV show!',
+    'Your app was opened more times than a blockbuster movie on its opening weekend!',
+  ],
+}
+
+// Check what treshold does the stat qualify for and return the fun comparison
+function getFunComparison(comparison: keyof typeof funComparisons, stat: number): string {
+  const thresholdsForComparisons = thresholds[comparison]
+  const index = thresholdsForComparisons.findIndex((threshold, index) => {
+    const thresholdGreaterThenStat = threshold >= stat;
+    const statGreaterOrEqualThresholdAndLastIndex = index === 2 && stat >= threshold;
+
+    return thresholdGreaterThenStat || statGreaterOrEqualThresholdAndLastIndex;
+  })
+
+  if (index === -1 || index >= 3)
+    throw new Error(`Cannot find index for fun comparison, ${index}`)
+
+  return funComparisons[comparison][index]
+}
+
+async function main(_url: URL, headers: BaseHeaders, _method: string, _body: any) {
+  const API_SECRET = getEnv('API_SECRET')
+  const authorizationSecret = headers.apisecret
+  if (!authorizationSecret)
+    return sendRes({ status: 'Cannot find authorization secret' }, 400)
+
+  if (!authorizationSecret || !API_SECRET || authorizationSecret !== API_SECRET) {
+    console.error('Fail Authorization', { authorizationSecret, API_SECRET })
```

Comment:
Can we log `API_SECRET` like this? Does this have no change of leaking?

Reply:
I have no idea. I just copied the template

Reply:
I am not touching the security of this, if there is a bug it there then it is already in production

Reply:
@ologbonowiwi 😖, you are right it was a local test who ended up in prod

Reply:
Should I remove the entire `console.error`?

Reply:
@riderx I removed the log from `supabase/functions/cron_good_plan/index.ts` on #368. I'm not sure if we should also remove the `authorizationSecret` please let me know if so

Reply:
Fixed this in `cron_email`